File: bench/node/pipeline.js
const N = parseInt(process.env.BENCH_N || "100000", 10);
function* range(n) { for (let i = 0; i < n; i++) yield i; }
const total = range(N)
  .filter((i) => i % 3 === 0 || i % 5 === 0)
  .map((i) => i * i)
  .reduce((a, b) => a + b, 0);
console.log(total);

File: bench/erlang/pipeline.erl
#!/usr/bin/env escript
%%! -mode(compile)
%% filter/map/reduce pipeline over a range: keep multiples of 3 or 5, square, sum.
main(_) ->
    N = bench_n(100000),
    Sum = pipe(0, N, 0),
    io:format("~w~n", [Sum]).
bench_n(D) -> case os:getenv("BENCH_N") of false -> D; V -> list_to_integer(V) end.
pipe(I, N, Acc) when I >= N -> Acc;
pipe(I, N, Acc) ->
    case (I rem 3 =:= 0) orelse (I rem 5 =:= 0) of
        true -> pipe(I+1, N, Acc + I*I);
        false -> pipe(I+1, N, Acc)
    end.
